setup/m2e.setup

```diff
@@ -168,44 +169,39 @@
         locateNestedProjects="true"/>
   </setupTask>
   <setupTask
-      xsi:type="setup:CompoundTask"
-      name="Working Sets">
-    <setupTask
-        xsi:type="setup.workingsets:WorkingSetTask"
-        prefix="org.eclipse.m2e.core.git-">
-      <workingSet
-          name="m2e-core">
-        <predicate
-            xsi:type="predicates:LocationPredicate"
-            pattern="${git.clone.m2e.core.location|path}"/>
-        <predicate
-            xsi:type="predicates:LocationPredicate"
-            pattern="${git.clone.m2e.core.location|path}/[^/]+"/>
-      </workingSet>
-      <description>The dynamic working sets for ${scope.project.label}</description>
-    </setupTask>
-    <setupTask
-        xsi:type="setup.workingsets:WorkingSetTask"
-        prefix="org.eclipse.m2e.core.git-">
-      <workingSet
-          name="m2e-maven-runtime">
-        <predicate
-            xsi:type="predicates:LocationPredicate"
-            pattern="${git.clone.m2e.core.location|path}/m2e-maven-runtime/[^/]+"/>
-      </workingSet>
-      <description>The dynamic working sets for ${scope.project.label}</description>
-    </setupTask>
-    <setupTask
-        xsi:type="setup.workingsets:WorkingSetTask"
-        prefix="org.eclipse.m2e.core.git-">
-      <workingSet
-          name="m2e-core-tests">
-        <predicate
-            xsi:type="predicates:LocationPredicate"
-            pattern="${git.clone.m2e.core.location|path}/m2e-core-tests/[^/]+"/>
-      </workingSet>
-      <description>The dynamic working sets for ${scope.project.label}</description>
-    </setupTask>
+      xsi:type="setup.workingsets:WorkingSetTask">
+    <workingSet
+        name="m2e-core"
+        id="m2e.core">
+      <predicate
+          xsi:type="predicates:AndPredicate">
+        <operand
+            xsi:type="predicates:RepositoryPredicate"
```

Comment:
@merks it looks like the `RepositoryPredicate` ignores sub-modules.
While this is correct, it would be sometimes convenient to be able to match projects from submodules too.
Do you think that would be possible?